db.go

```diff
@@ -31,6 +31,14 @@ const (
 	GetIdentitiesFlagDeleted GetIdentitiesFlag = 1 << (iota - 1)
 )
 
+type AuthCheck int
```

Comment:
Sorry to create more issues, but these things MUST be in separate enums. They don't share any functionality.
One of them is AuthCheck, and the other one is PasswordEnforcement.
Also, I would rename AuthCheckNone to AuthCheckDefault, because None implies that we're not even checking the password.